router/cmd/plan-generator.go

```
@@ -0,0 +1,147 @@
+	_ = godotenv.Load()
+	err := env.Parse(&cfg)
+	if err != nil {
+		return cfg, err
+	}
+	return cfg, nil
+}
+
+func planGeneratorCmd() error {
+	cfg, err := getParseGeneratorConfig()
+	if err != nil {
+		return err
+	}
+
+	return PlanGenerator(cfg)
+}
+
+func PlanGenerator(cfg QueryPlanConfig) error {
+	cfg, err := getParseGeneratorConfig()
+	if err != nil {
+		return err
+	}
+
+	queriesPath, err := filepath.Abs(cfg.SourceDir)
+	if err != nil {
+		return fmt.Errorf("failed to get absolute path for queries: %v", err)
+	}
+
+	outPath, err := filepath.Abs(cfg.OutDir)
+	if err != nil {
+		return fmt.Errorf("failed to get absolute path for output: %v", err)
+	}
+	if err := os.MkdirAll(outPath, 0755); err != nil {
+		log.Fatalf("failed to create output directory: %v", err)
+	}
+
+	executionConfigPath, err := filepath.Abs(cfg.ExecutionConfig)
+	if err != nil {
+		return fmt.Errorf("failed to get absolute path for execution config: %v", err)
+	}
+
+	var filter []string
+	if cfg.Filter != "" {
+		filterContent, err := os.ReadFile(cfg.Filter)
+		if err != nil {
+			return fmt.Errorf("failed to read filter file: %v", err)
+		}
+
+		filter = strings.Split(string(filterContent), "\n")
+	}
+
+	queries, err := os.ReadDir(queriesPath)
+	if err != nil {
+		return fmt.Errorf("failed to read queries directory: %v", err)
+	}
+
+	queriesQueue := make(chan os.DirEntry, len(queries))
+	for _, queryFile := range queries {
+		queriesQueue <- queryFile
+	}
+	close(queriesQueue)
+
+	duration, parseErr := time.ParseDuration(cfg.Timeout)
+	if parseErr != nil {
+		return fmt.Errorf("failed to parse timeout: %v", parseErr)
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), duration)
+	defer cancel()
+
+	t := time.Now()
+	wg := sync.WaitGroup{}
+	wg.Add(cfg.Concurrency)
+	for i := 0; i < cfg.Concurrency; i++ {
+		go func(i int) {
+			defer wg.Done()
+			pg, err := core.NewPlanGenerator(executionConfigPath)
+			if err != nil {
+				log.Fatalf("failed to create plan generator: %v", err)
```

One error should not fail the entire procedure. We should output a nice table view of the errors. We could later add a `--fail-fast` option.

---

I think that if the NewPlanGenerator fails, we probably will have everthing failing, so failing fast could be a good strategy. When the single query fails we only print the error (https://github.com/wundergraph/cosmo/pull/1627/files#diff-caa4e23b1af0c5290d7e7c2017a433c07a8aad1d078d244c856718f994a13f6eR129) and output an empty file.

---

We also need to render the plan error somewhere. Ideally, we can think of a specification that contains the plan and potential errors

```json
{ "plan": "...", "error": "..." }
```

---

I've added a results.jsonl file in the output folder with the following format:
```jsonl
{"file_name": "example.graph", "plan": "whole plan...", "error":"error, if any is returned from the planner"}
{"file_name": "example2.graph", "plan": "whole plan...", "error":"error, if any is returned from the planner"}
```